### PR TITLE
unit_tests: use valid key images, pub keys, etc

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

**Comment by danrmiller** [comment]
Getting "Wrong result" on the cncrypto tests.

https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/4856/steps/test/logs/LastTest

**Comment by moneromooo-monero** [comment]
Those are the ones that are dependent on the exact determinism of the PRNG, and master has a patch which changes the generation of 32 byte values to remove a slight bias. The tests are fixed in my patch to change the RNG to use libsodium. If it doesn't go in, I'll split the tests part off and PR it.

**Comment by moneromooo-monero** [comment]
aka, fixed if you also apply #4159.

File: tests/unit_tests/ringdb.cpp [hunk 1]
```diff
@@ -39,25 +39,29 @@
...
 #include "wallet/ringdb.h"
 
 static crypto::chacha_key generate_chacha_key()
 {
-  uint8_t key[CHACHA_KEY_SIZE];
-  crypto::rand(CHACHA_KEY_SIZE, key);
   crypto::chacha_key chacha_key;
-  memcpy(&chacha_key, key, CHACHA_KEY_SIZE);
+  uint64_t password = crypto::rand<uint64_t>();
+  crypto::generate_chacha_key(std::string((const char*)&password, sizeof(password)), chacha_key);
```
Comment (MiWCryptoCurrency) [inline tests/unit_tests/ringdb.cpp]: appears to be missing the number of KDF rounds which breaks test

**Comment by MiWCryptoCurrency** [comment]
Is there something wrong with the test in ringdb.cpp:50? 
It appears to be missing the number of kdf rounds, im getting:

monero/tests/unit_tests/ringdb.cpp: In function ‘crypto::chacha_key generate_chacha_key()’:
monero/tests/unit_tests/ringdb.cpp:50:96: error: no matching function for call to ‘generate_chacha_key(std::__cxx11::string, crypto::chacha_key&)’
   crypto::generate_chacha_key(std::string((const char*)&password, sizeof(password)), chacha_key);

monero/src/crypto/chacha.h:90:15: note: candidate: ‘void crypto::generate_chacha_key(std::__cxx11::string, crypto::chacha_key&, uint64_t)’
   inline void generate_chacha_key(std::string password, chacha_key& key, uint64_t kdf_rounds) {
               ^~~~~~~~~~~~~~~~~~~
monero/src/crypto/chacha.h:90:15: note:   candidate expects 3 arguments, 2 provided

**Comment by MiWCryptoCurrency** [comment]
Seem the change to add the KDF param was added in PR #4103
Need to modify the test to add the KDF param -- this comes from wallet2.h, right?

**Comment by stoffu** [comment]
Thanks for reporting, #4272

**Pull request merged**